sklearn/tests/test_random_projection.py

```diff
@@ -377,6 +378,51 @@ def test_random_projection_feature_names_out(random_projection_cls):
     assert_array_equal(names_out, expected_names_out)
 
 
+@pytest.mark.parametrize("random_state", range(50))
+@pytest.mark.parametrize("n_rows", (2, 9, 10, 11, 1000))
+@pytest.mark.parametrize("n_cols", (2, 9, 10, 11, 1000))
+@pytest.mark.parametrize("random_projection_cls", all_RandomProjection)
+def test_inverse_transform(random_state, n_rows, n_cols, random_projection_cls):
+    n_components = 10
+    for data in make_sparse_random_data(
+        n_rows, n_cols, n_rows * n_cols // 100 + 1, random_state
+    ):
+        random_projection = random_projection_cls(
+            n_components=n_components, fit_inverse_transform=True
+        )
+        if n_cols < n_components:
+            with pytest.warns(
+                DataDimensionalityWarning,
+                match="The number of components is higher than the number of features",
+            ):
+                projected = random_projection.fit_transform(data)
+        else:
+            projected = random_projection.fit_transform(data)
+        projected_back = random_projection.inverse_transform(projected)
+        assert projected_back.shape == data.shape
+        projected_again = random_projection.transform(projected_back)
+        if hasattr(projected, "toarray"):
+            projected = projected.toarray()
+        assert_array_almost_equal(projected, projected_again)
```

Comment:
please use `assert_allclose`

Reply:
Sounds good, I'll do that

Reply:
Hmm, for some reason it passes with `assert_array_almost_equal()` but not with `assert_allclose()`, I'm not sure why. Tbh I don't have much time right now, I'm trying to finish writing the 3rd edition of my book, so if a kind soul wants to investigate this point, that would be great. What is the difference between `assert_array_almost_equal()` and `assert_allclose()`? May the difference be about sparse vs dense?

Reply:
The difference is that `assert_array_almost_equal` checks the absolute difference between 2 arrays while `assert_allclose` checks the relative difference. We don't want the quality of the test to depend on the magnitude of the arrays.

>  Tbh I don't have much time right now

Don't worry, I'll look into it.

Reply:
The issue is that for the SparseRandomProjection, the transformed data is expected to be sparse and thus have true zeros. When you project back and project again, these zeros can become a very small non-zero value. The relative difference between 0 and any non-zero value is always 100%, thus the failure. We need to set, in addition to the relative tol, an absolute tol. Absolute tol can be seen as "any value below abs_tol is considered to be 0".

I directly pushed some changes in the test